cfdm/data/subarray/mixin/biquadraticlatitudelongitudeinterpolation.py

```diff
@@ -0,0 +1,242 @@
+import numpy as np
+
+from .quadraticgeographicinterpolation import QuadraticGeographicInterpolation
+
+
+class BiQuadraticLatitudeLongitudeInterpolation(
+    QuadraticGeographicInterpolation
+):
+    """Mixin class for bi-quadratic latitude longitude interpolation.
+
+    See CF appendix J "Coordinate Interpolation Methods".
+
+    .. versionadded:: (cfdm) 1.9.TODO.0
+
+    """
+
+    def _bi_quadratic_latitude_longitude_interpolation(
+        self,
+        lat_a,
+        lon_a,
+        lat_b,
+        lon_b,
+        lat_c,
+        lon_c,
+        lat_d,
+        lon_d,
+        ce1,
+        ca1,
+        ce2,
+        ca2,
+        ce3,
+        ca3,
+        location_use_3d_cartesian,
+        d2,
+        d1,
+    ):
+        """Bi-quadratic interpolation of geographic coordinates.
+
+        A two-dimensional quadratic method for interpolation of the
+        geographic coordinates latitude and longitude, typically used
+        for remote sensing products with geographic coordinates on the
+        reference ellipsoid.
+
+        See CF appendix J "Coordinate Interpolation Methods".
+
+        .. versionadded:: (cfdm) 1.9.TODO.0
+
+        :Parameters:
+
+            lat_a, lon_a: `numpy.ndarray`
+                The latitude and longitude of the first tie point in
+                index space of subsampled dimension 1 (in the sense of
+                CF appendix J Figure J.1).
```

Comment:
It's not clear to me which figures are being referred to here, since cross-referencing the current document at '[Latest release of CF conventions](https://cfconventions.org/Data/cf-conventions/cf-conventions-1.9/cf-conventions.html)', the diagrams in appendix J are numbered from 7 to 11?

Reply:
Ah yes. The figure numbering was a defect in the release of CF-1.9. This has now been fixed (https://github.com/cf-convention/cf-conventions/issues/352), and is OK (i.e. J.1, etc.) in the draft of CF-1.10 (https://cfconventions.org/cf-conventions/cf-conventions.html#appendix-coordinate-subsampling)